src/Raven.Server/Documents/Sharding/ShardedDocumentsStorage.cs

```diff
@@ -39,6 +38,69 @@ public IEnumerable<Document> GetDocumentsByBucketFrom(DocumentsOperationContext
         }
     }
 
+    public static IEnumerable<BucketStatistics> GetBucketStatistics(DocumentsOperationContext context, int start, int take = int.MaxValue)
+    {
+        var tree = context.Transaction.InnerTransaction.ReadTree(BucketStatsSlice);
+        if (tree == null)
+            yield break;
+
+        using (GetBucketByteString(context.Allocator, start, out var buffer))
+        using (Slice.External(context.Allocator, buffer, buffer.Length, out var startSlice))
+        {
+            using (var it = tree.Iterate(prefetch: true))
+            {
+                if (it.Seek(startSlice) == false)
+                    yield break;
+
+                do
+                {
+                    if (take-- <= 0)
+                        yield break;
+
+                    var reader = it.CreateReaderForCurrent();
+                    yield return ValueReaderToBucketStats(it.CurrentKey, ref reader);
+
+                } while (it.MoveNext());
+            }
+        }
+    }
+
+    private static BucketStatistics ValueReaderToBucketStats(Slice key, ref ValueReader valueReader)
+    {
+        var bucket = Bits.SwapBytes(*(int*)key.Content.Ptr);
+        var stats = *(Voron.Data.BucketStats*)valueReader.Base;
+        return new BucketStatistics
+        {
+            Bucket = bucket,
+            Size = stats.Size,
+            NumberOfItems = stats.NumberOfDocuments,
+            LastModified = new DateTime(stats.LastModifiedTicks)
```

Comment:
let's specify here and on all other places the `Utc` kind, just to be on the safe side

Reply:
done